sdk/python/kfp/dsl/_pipeline.py

```diff
@@ -19,6 +19,11 @@
 import sys
 
 
+# Contains a stack of handler functions.
```

Comment:
Could you explain why we need the decorator handlers?

Reply:
This is a bit similar to the "dependency injection" design pattern.

Right know, the `@pipeline` decorator is always running some code that is only needed in a specific compiler scenario. The decorator has a dependency on that compiler-related code.
I'm removing that dependency.

But during some command-line compilation scenarios the code still needs to be run. So I'm adding a way for the command-line compiler to add a handler and run the code it wants (but only when it's needed, not every time).
That's how we enable the `@pipeline` to run the command-line compiler specific code without introducing a dependency.

Reply:
Got it. Thanks for the explanation.